#!/usr/bin/env python2
PACKAGE = "event_ros_tools"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("rate_averaging_time", double_t, 0,
        "time [s] for exponential moving rate averaging",
        0.5, 0.05, 10.0)

gen.add("lead_time", double_t, 0,
        "time [s] to start integration before event burst",
        2e-3, -5e-3, 0.1)

gen.add("duration", double_t, 0,
        "time [s] for which to integrate events",
        5e-3, 1e-4, 50e-3)

gen.add("avoid_tearing", bool_t, 0,
        "keep integrating events until frame complete",
        True)

gen.add("use_on_events", bool_t, 0,
        "whether to use ON or OFF events for image", True)

gen.add("denoise", bool_t, 0,
        "whether to denoise image", True)

gen.add("denoise_num_frames", int_t, 0,
        "how many frames to use for denoising", 1, 1, 5)

gen.add("denoise_template_window_size", int_t, 0,
        "template window size for denoising", 7, 3, 9)

gen.add("denoise_search_window_size", int_t, 0,
        "search window size for denoising", 21, 7, 31)

gen.add("denoise_h", double_t, 0,
        "h parameter for denoising", 20.0, 1, 100.0)

exit(gen.generate(PACKAGE, "event_ros_tools", "FlickerDyn"))
